### PR TITLE
Fix division fix delay execution

### PR DESCRIPTION
**Pull request by Rajkumar-D**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java [hunk 1]
```diff
@@ -86,6 +82,39 @@ public ValidationResult check(Entry entry) throws ValidationEngineException {
...
+            List<Taxon> taxonList;
+            if (!(taxonList = new TaxonomyClientImpl().getTaxonByScientificName(primarySF.getScientificName())).isEmpty()) {
+                division = taxonList.get(0).getDivision();
+                saveDivisionCache(primarySF.getScientificName(), division);
+            }
+        }
+        return division;
+    }
+    
+    private String getDivisionFromCache(String divisionKey){
```
Comment (raskoleinonen) [inline src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:110]: - Minor name change suggestion: divisionKey -> scientificName
- Code simplification suggestion:
```
if (scientificName == null) {
 return null;
}
return EmblEntryValidationPlan.divisionCache.get(scientificName);
```

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java [hunk 2]
```diff
@@ -86,6 +82,39 @@ public ValidationResult check(Entry entry) throws ValidationEngineException {
...
+        return EmblEntryValidationPlan.divisionCache.get(divisionKey)!=null ?  EmblEntryValidationPlan.divisionCache.get(divisionKey) : null;
+    }
+    
+    private void saveDivisionCache(String divisionKey, String division){
```
Comment (raskoleinonen) [inline src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:114]: - Minor name change suggestion: divisionKey -> scientificName

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java [hunk 3]
```diff
@@ -86,6 +82,39 @@ public ValidationResult check(Entry entry) throws ValidationEngineException {
...
+            division = new TaxonomyClientImpl().getTaxonByTaxid(primarySF.getTaxId()).getDivision();
+            saveDivisionCache(primarySF.getTaxId().toString(), division);
+        }
+        return division;
+    }
+    
+    private String getDivisionByScientificName(SourceFeature primarySF){
+        String division = "";
+        if(primarySF.getScientificName()!=null && getDivisionFromCache(primarySF.getScientificName())!=null){
+            division = getDivisionFromCache(primarySF.getScientificName());
```
Comment (raskoleinonen) [inline src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:99]: - Code simplification suggestion (with changes suggested below):
```
String division = getDivisionFromCache(primarySF.getScientificName());
if (empty(division)) {
    List<Taxon> taxonList;
    ....
```

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java (hunk 1 above)
Comment (Rajkumar-D) [reply src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:110]: Name suggestion (divisionKey -> scientificName)  is not added as the same method is used for used for taxId and scientificName 
Code simplification suggestion is added

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java (hunk 2 above)
Comment (Rajkumar-D) [reply src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:114]: Name suggestion (divisionKey -> scientificName) is not added as the same method is used for used for taxId and scientificName

File: src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java (hunk 3 above)
Comment (Rajkumar-D) [reply src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java:99]: Code simplification suggestion is added

**Pull request merged**